case_conversion/case_parse.py

```diff
@@ -7,75 +7,199 @@
     unicode = str
 
 
-def parse_case(var, detect_acronyms=True, acronyms=[], preserve_case=False):
-    """
-    Parses a variable into a list of words.
-    Also returns the case type, which can be one of the following:
-        - upper: All words are upper-case.
-        - lower: All words are lower-case.
-        - pascal: All words are title-case or upper-case. Note that the
-                  variable may still have separators.
-        - camel: First word is lower-case, the rest are title-case or
-                 upper-case. Variable may still have separators.
-        - mixed: Any other mixing of word casing. Never occurs if there are
-                 no separators.
-        - unknown: Variable contains no words.
-    Also returns the first separator character, or False if there isn't one.
+UPPER = regex.compile(u'^[\p{Lu}]$')
```

Comment:
I totally agree `regex` being superior over `re`.
Still, is it worthwhile to add a new dependency here?